src/Winton.Extensions.Configuration.Consul/IConsulConfigurationClient.cs

```diff
@@ -12,12 +13,16 @@ namespace Winton.Extensions.Configuration.Consul
     internal interface IConsulConfigurationClient
     {
         /// <summary>Gets the config from consul asynchronously.</summary>
+        /// <param name="key">The key at which the config is located.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used to cancel the operation.</param>
         /// <returns>A task containing the result of the query for the config.</returns>
-        Task<QueryResult<KVPair>> GetConfig();
+        Task<QueryResult<KVPair[]>> GetConfig(string key, CancellationToken cancellationToken);
 
-        /// <summary>Watches the config for changes.</summary>
+        /// <summary>Watches for config changes at a specified key.</summary>
+        /// <param name="key">The key whose value should be watched for changes.null</param>
```

Comment:
Maybe a stupid question, but does that `null` at the end there do something magic? Or is it a typo?

Reply:
Definite typo. I shall fix.